bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/discovery/ShellyDiscoveryParticipant.java

```diff
@@ -174,7 +173,7 @@ public DiscoveryResult createResult(final ServiceInfo service) {
                     thingUID = ShellyDeviceProfile.getThingUID(name, mode, true);
                 } else {
                     logger.info("{}: {}", name, messages.get("discovery.failed", address));
-                    logger.debug("{}: Exception {}", name, e.toString());
+                    logger.debug("{}: {}", name, e.toString());
```

Comment:
same here:
```suggestion
                    logger.debug("{}:", name, e);
```

Reply:
example:
```
ERROR] Code Analysis Tool has found:
4 error(s)!
9 warning(s)
10 info(s)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[158]
Count of placeholder(2) is not equal to count of parameter(1)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[325]
Count of placeholder(3) is not equal to count of parameter(2)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.java:[392]
Count of placeholder(3) is not equal to count of parameter(2)
[ERROR] org.openhab.binding.shelly.internal.handler.ShellyBaseHand
```

Reply:
You should notice that when adding an exception as the last argument you don't include a `{}` for it in the log format string. 

Notice that the code analysis didn't complain about line 171 😄

Reply:
done